lib/websock.ex

```diff
@@ -57,7 +57,7 @@ defmodule WebSock do
           {:push, {opcode(), message()} | [{opcode(), message()}], state()}
           | {:reply, term(), {opcode(), message()} | [{opcode(), message()}], state()}
           | {:ok, state()}
-          | {:stop, term(), state()}
+          | {:stop, {:shutdown, :draining} | term(), state()}
```

Comment:
Maybe this is more generic?

```suggestion
          | {:stop, {:shutdown, :restart} | term(), state()}
```